File: csharp/grade-school/GradeSchool.cs
using System.Linq;
using System.Collections.Generic;

public class GradeSchool
{
    private List<(string Name, int Grade)> students = 
        new List<(string Name, int Grade)>();

    public void Add(string student, int grade)
    {
        students.Add((student, grade));
        students.Sort((x, y) => x.Grade == y.Grade
            ? string.Compare(x.Name, y.Name)
            : x.Grade.CompareTo(y.Grade));
    }

    public IEnumerable<string> Roster() =>
        students.Select(x => x.Name);

    public IEnumerable<string> Grade(int grade) =>
        students.Where(x => x.Grade == grade).Select(x => x.Name);
}

File: python/grade-school/grade_school.py
# Default dict removes the issue of inserting keys for the first time vs appending items
# as well as returning an empty list (by initializing it with a list argument)
# for a key that doesn't exist
from collections import defaultdict


class School:
    def __init__(self):
        self.students = defaultdict(list)

    def add_student(self, name, grade):
        self.students[grade].append(name)
        self.students[grade].sort()

    def roster(self):
        return [name for k, values in sorted(self.students.items()) for name in values]

    def grade(self, grade):
        return self.students[grade]
